src/backend/distributed/transaction/citus_dist_stat_activity.c

```diff
@@ -154,12 +154,50 @@
 		pg_stat_activity.query, \
 		pg_stat_activity.backend_type \
 	FROM \
-		get_all_active_transactions() AS dist_txs(database_id, process_id, initiator_node_identifier, worker_query, transaction_number, transaction_stamp), \
 		pg_stat_activity \
+		LEFT JOIN \
+		get_all_active_transactions() AS dist_txs(database_id, process_id, initiator_node_identifier, worker_query, transaction_number, transaction_stamp) \
+		ON pg_stat_activity.pid = dist_txs.process_id \
 	WHERE \
-		pg_stat_activity.pid = dist_txs.process_id \
-			AND \
-		dist_txs.worker_query = %s; "
+		(dist_txs.worker_query IS NULL OR dist_txs.worker_query = false) \
+		AND \
+		pg_stat_activity.query NOT ILIKE '%stat_activity%';"
```

Comment:
Can we add `AND application_name = 'citus'`? we're not so interested in background workers and such.

Reply:
I have added that check for worker stat activities, do we need to add it for dist stat activities as well?